client/src/components/Project/ProjectListing.js

```diff
@@ -0,0 +1,182 @@
+import Table from '../Table/Table.js'
+import { useSelector, useDispatch } from 'react-redux'
+import Card from '../Card/Card.js'
+import CardHeader from '../Card/CardHeader.js'
+import CardBody from '../Card/CardBody.js'
+import React, { useState, useEffect } from 'react'
+// import GridItem from '../../components/Grid/GridItem.js'
+import { makeStyles } from '@material-ui/core/styles'
+import Button from '../CustomButtons/Button.js'
+import styles from '../../assets/jss/material-dashboard-react/views/dashboardStyle'
+import GridItem from '../Grid/GridItem.js'
+// import DialogActions from '@material-ui/core/DialogActions';
+import { Dialog, DialogActions } from '@material-ui/core'
+import { deleteProject, clearProjectMsg } from '../../actions/projectAction'
+import { useToasts } from 'react-toast-notifications'
+import Employee from './Project'
+const useStyles = makeStyles(styles)
+const ProjectListing = props => {
+  const { projectData, setPageView } = props
+  const { addToast } = useToasts()
+  const classes = useStyles()
+  const dispatch = useDispatch()
+  const [projectToUpdate, setProjectToUpdate] = useState()
+  const [updateAction, setUpdateAction] = useState()
+  const [showDelDialog, setShowDelDialog] = useState(false)
+  const deleteProjectSuccess = useSelector(
+    state => state.projectReducer.deleteProjectSuccess
+  )
+  const deleteProjectError = useSelector(
+    state => state.projectReducer.deleteProjectError
+  )
+
+  const projectListingHeader = [
+    'Title',
+    'Technology',
+    'Client',
+    'Client Location',
+    'Start Date',
+    'End Date'
+  ]
+
+  useEffect(() => {
+    if (deleteProjectSuccess) {
+      addToast(deleteProjectSuccess, {
+        appearance: 'success',
+        autoDismiss: true
+      })
+      dispatch(clearProjectMsg())
+    }
+  }, [deleteProjectSuccess, addToast, dispatch])
+
+  useEffect(() => {
+    if (deleteProjectError) {
+      addToast(deleteProjectError, {
+        appearance: 'error',
+        autoDismiss: true
+      })
+      dispatch(clearProjectMsg())
+    }
+  }, [deleteProjectError, addToast, dispatch])
+
+  const projectDetails = []
+  let filteredProject
+  if (projectData) {
+    filteredProject = projectData.filter(cls => cls.status !== 'Inactive')
+    filteredProject.map((key, value) => {
+      let {
+        title,
+        technology,
+        client,
+        client_location,
+        startdate,
+        enddate
+      } = key
+
+      const start = new Date(startdate)
+      const end = new Date(enddate)
+      const mnth = ('0' + (start.getMonth() + 1)).slice(-2)
```

Comment:
Can you please move date logic to date helper? this can be used at multiple places.

Reply:
Added in helper function